src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -129,112 +183,570 @@ def __repr__(self):
+
+        Return:
+            FeaturePercentileAllocation object
+        """
+        percentile_allocation = json.loads(json_string)
+        variant = percentile_allocation.get(FeatureFlagConstants.VARIANT, None)
+        percentileFrom = percentile_allocation.get(FeatureFlagConstants.FROM, None)
+        perecentileTo = percentile_allocation.get(FeatureFlagConstants.TO, None)
+        if not variant or not percentileFrom or not perecentileTo:
+            raise ValidationError(
+                "Percentile allocation must contain required '%s', '%s' and '%s' attributes: \n%s"
+                % (FeatureFlagConstants.VARIANT, FeatureFlagConstants.TO, FeatureFlagConstants.FROM, json.dumps(percentile_allocation, indent=2, ensure_ascii=False))
+            )
+
+        if not isinstance(percentileFrom, int) or not isinstance(perecentileTo, int):
+            raise ValidationError(
+                "Percentile allocation '%s' and '%s' must be integers: \n%s"
+                % (FeatureFlagConstants.FROM, FeatureFlagConstants.TO, json.dumps(percentile_allocation, indent=2, ensure_ascii=False))
+            )
+
+        if (
+            percentileFrom < 0 or
+            percentileFrom > 100 or
+            perecentileTo < 0 or
+            perecentileTo > 100
+        ):
+            raise ValidationError(
+                "Percentile allocation '%s' and '%s' must be between 0 and 100: \n%s"
+                % (FeatureFlagConstants.TO, FeatureFlagConstants.FROM, json.dumps(percentile_allocation, indent=2, ensure_ascii=False))
+            )
+
+        if percentileFrom >= perecentileTo:
+            raise ValidationError(
+                "Percentile allocation '%s' must be less than '%s': \n%s"
+                % (FeatureFlagConstants.FROM, FeatureFlagConstants.TO, json.dumps(percentile_allocation, indent=2, ensure_ascii=False))
+            )
+
+        return cls(variant=variant, From=percentileFrom, to=perecentileTo)
+
+    def __repr__(self):
+        feature_percentile_allocation = {
+            FeatureFlagConstants.VARIANT: self.variant,
+            FeatureFlagConstants.FROM: self.From,
+            FeatureFlagConstants.TO: self.to,
+        }
+
+        return json.dumps(feature_percentile_allocation, indent=2, ensure_ascii=False)
+
+
+class FeatureAllocation:
+    """
+    Feature allocation class.
+
+    :ivar list FeatureUserAllocation user:
+        Determines how variants should be allocated for the feature to various users.
+    :ivar list FeatureGroupAllocation group:
+        Determines how variants should be allocated for the feature to various groups.
+    :ivar list FeaturePercentileAllocation percentile:
+        Determines how variants should be allocated for the feature to various users based on percentile.
+    """
+
+    def __init__(
+        self,
+        user=None,
+        group=None,
+        percentile=None,
+        default_when_enabled=None,
+        default_when_disabled=None,
```

Comment:
Should we add the `seed` property for allocation?

Reply:
Yes we should. Updated